book/internals.rst

```diff
@@ -1,666 +0,0 @@
-    response won't work. If you want to overwrite the status code (which you
-    should not without a good reason), set the ``X-Status-Code`` header::
-
-        return new Response(
-            'Error',
-            404 // ignored,
-            array('X-Status-Code' => 200)
-        );
-
-.. seealso::
-
-    Read more on the :ref:`kernel.exception event <component-http-kernel-kernel-exception>`.
-
-.. index::
-   single: EventDispatcher
-
-The EventDispatcher
--------------------
-
-The EventDispatcher is a standalone component that is responsible for much
-of the underlying logic and flow behind a Symfony request. For more information,
-see the :doc:`EventDispatcher component documentation </components/event_dispatcher/introduction>`.
-
-.. index::
-   single: Profiler
-
-.. _internals-profiler:
-
-Profiler
---------
-
-When enabled, the Symfony profiler collects useful information about each
-request made to your application and store them for later analysis. Use the
-profiler in the development environment to help you to debug your code and
-enhance performance; use it in the production environment to explore problems
-after the fact.
-
-You rarely have to deal with the profiler directly as Symfony provides
-visualizer tools like the Web Debug Toolbar and the Web Profiler. If you use
-the Symfony Standard Edition, the profiler, the web debug toolbar, and the
-web profiler are all already configured with sensible settings.
-
-.. note::
-
-    The profiler collects information for all requests (simple requests,
-    redirects, exceptions, Ajax requests, ESI requests; and for all HTTP
-    methods and all formats). It means that for a single URL, you can have
-    several associated profiling data (one per external request/response
-    pair).
-
-.. index::
-   single: Profiler; Visualizing
-
-Visualizing Profiling Data
-~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Using the Web Debug Toolbar
-...........................
-
-In the development environment, the web debug toolbar is available at the
-bottom of all pages. It displays a good summary of the profiling data that
-gives you instant access to a lot of useful information when something does
-not work as expected.
-
-If the summary provided by the Web Debug Toolbar is not enough, click on the
-token link (a string made of 13 random characters) to access the Web Profiler.
-
-.. note::
-
-    If the token is not clickable, it means that the profiler routes are not
-    registered (see below for configuration information).
-
-Analyzing Profiling Data with the Web Profiler
-..............................................
-
-The Web Profiler is a visualization tool for profiling data that you can use
-in development to debug your code and enhance performance; but it can also be
-used to explore problems that occur in production. It exposes all information
-collected by the profiler in a web interface.
-
-.. index::
-   single: Profiler; Using the profiler service
-
-Accessing the Profiling information
```

Comment:
do we have this convered somewhere? If not, we should create a cookbook article (or add it to an existing one).